app/src/main/java/org/p2p/wallet/settings/ui/settings/presenter/NewSettingsPresenter.kt

```diff
@@ -0,0 +1,125 @@
+package org.p2p.wallet.settings.ui.settings.presenter
+
+import android.content.Context
+import org.p2p.solanaj.rpc.NetworkEnvironment
+import org.p2p.wallet.R
+import org.p2p.wallet.auth.analytics.AdminAnalytics
+import org.p2p.wallet.auth.interactor.AuthInteractor
+import org.p2p.wallet.auth.interactor.AuthLogoutInteractor
+import org.p2p.wallet.auth.interactor.UsernameInteractor
+import org.p2p.wallet.common.AppRestarter
+import org.p2p.wallet.common.crypto.keystore.EncodeCipher
+import org.p2p.wallet.common.mvp.BasePresenter
+import org.p2p.wallet.home.analytics.BrowseAnalytics
+import org.p2p.wallet.home.repository.HomeLocalRepository
+import org.p2p.wallet.infrastructure.network.environment.NetworkEnvironmentManager
+import org.p2p.wallet.receive.analytics.ReceiveAnalytics
+import org.p2p.wallet.renbtc.service.RenVMService
+import org.p2p.wallet.settings.interactor.SettingsInteractor
+import org.p2p.wallet.settings.ui.settings.NewSettingsContract
+import timber.log.Timber
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+
+private const val NETWORK_CHANGE_DELAY = 250L
+
+class NewSettingsPresenter(
+    private val environmentManager: NetworkEnvironmentManager,
+    private val usernameInteractor: UsernameInteractor,
+    private val authLogoutInteractor: AuthLogoutInteractor,
+    private val appRestarter: AppRestarter,
+    private val receiveAnalytics: ReceiveAnalytics,
+    private val adminAnalytics: AdminAnalytics,
+    private val browseAnalytics: BrowseAnalytics,
+    private val settingsInteractor: SettingsInteractor,
+    private val homeLocalRepository: HomeLocalRepository,
+    private val settingsItemMapper: SettingsItemMapper,
+    private val authInteractor: AuthInteractor,
+    private val context: Context
+) : BasePresenter<NewSettingsContract.View>(), NewSettingsContract.Presenter {
+
+    override fun attach(view: NewSettingsContract.View) {
+        super.attach(view)
+        loadSettings()
+    }
+
+    private fun loadSettings() {
+        try {
```

Comment:
what we can handle here ?

Reply:
interactor and SecureStorage inside **can** throw some stuff